Automattic/newspack-plugin
# feat(ras-acc): add standalone newsletter signup form

### All Submissions:






### Changes proposed in this Pull Request:



Closes https://app.asana.com/0/1206943664367857/1205668937700578/f

This PR is paired with the following blocks PR: https://github.com/Automattic/newspack-blocks/pull/1737

This PR separates the post-checkout newsletters signup form from the checkout thank you modal.

This PR also moves logic for rendering and handling the newsletters signup form from the blocks plugin over to the main Newspack plugin since the form only renders whenever RAS is active. (Let me know if there is a preference to move this back to blocks)

![Screenshot 2024-05-20 at 16 51 15](https://github.com/Automattic/newspack-plugin/assets/17905991/00977454-73a9-410d-bedb-2722965acfbd)

### How to test the changes in this Pull Request:

Before testing, be sure to also checkout the changes in https://github.com/Automattic/newspack-blocks/pull/1737

- Enable Newsletter Subscription Lists via the Newspack > Engagement > Show Advanced Settings menu and ensure at least two lists are available. This will enable the post-checkout newsletter signup form:

![Screenshot 2024-05-21 at 13 40 29](https://github.com/Automattic/newspack-plugin/assets/17905991/70d29a67-7c17-4c89-837b-d72b759315e9)

- As a reader that is not subscribed to at least one of your available newsletter lists, trigger the modal checkout flow via any donation or checkout button block (you will need to add one of these blocks to any page or post if you haven't already)
- Confirm the newsletter signup form appears only after you click the confirmation button on the thank you modal. So the flow after clicking the donation or checkout button should be:
    1. If not signed in, auth modal
    2. Checkout details/confirmation modal
    3. Thank you page modal
    4. Finally newsletter signup modal
- As admin, change the after success behavior to custom URL:

![Screenshot 2024-05-21 at 13 36 31](https://github.com/Automattic/newspack-plugin/assets/17905991/0222afe8-8b0c-47b0-af1b-b0ae44a55fe9)

- Repeat the checkout flow, again as a reader not subscribed to at least one of your available newsletter lists, but this time ensure you are correctly redirected after the newsletter signup modal step (be sure to subscribe to all remaining lists this time around for the next step of testing).
- Repeat the checkout flow, as a reader that is subscribed to all of your available newsletter lists. This time confirm the newsletters modal does not appear, but the post checkout after success behavior still works as expected.
- Finally, disable the Newsletter Subscription Lists setting altogether.
- Repeat the checkout flow and confirm the newsletters modal does not appear, and the after success behavior still works as expected.

### Other information:
